vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -652,7 +654,7 @@ base_learner* ccb_explore_adf_setup(options_i& options, vw& all)
     options.add_and_parse(new_options);
   }
 
-  if (!options.was_supplied("cb_sample"))
+  if (!options.was_supplied("cb_sample") && !do_not_sample)
```

Comment:
If `cb_sample` was supplied, then `do_not_sample` is ignored.
This behavior is not very user friendly. Maybe you could throw or log a warning in that case?

Reply:
Good point, when cb_sample is supplied explicitly it overrides this.

This option prevents CCB from adding the automatic cb_sample option. But if cb_sample is supplied in the model explicitly then it will also be a problem. Perhaps I should check this option in the cb_sample reduction itself?

Reply:
What is the scope of this option: global or CCB only?
If it is CCB only, i suggest to do it in the ccb_explore_adf reduction.
If it is global (CCB + CB), then it makes sense to do it in the cb_sample reduction.

Reply:
Yes it depends on that. If it is CCB only, but the model contains the cb_sample flag, then do_not_sample will not do as intended. It either needs to remove the flag from the options or operate in the cb_sample reduction. I am leaning towards the latter as it is less complicated code with, and making this a global thing